Remotion/ObjectBinding/Web.Development.WebTesting/ControlObjects/BocAutoCompleteReferenceValueControlObject.cs

```diff
@@ -332,25 +332,34 @@ private SearchServiceResultItem GetFirstAutoCompleteResult ([NotNull] string fil
   data = Sys.Serialization.JavaScriptSerializer.serialize(data);
 
   var returnValue = null;
-  var request = {{
-    async:false,
-    type:'POST',
-    contentType:'application/json; charset=utf-8',
-    url:options.serviceUrl + '/' + options.{3},
-    data:data,
-    dataType:'json',
-    success:function(result, context, methodName){{ returnValue = {{ {4}:'{6}', {5}:result.d }}; }},
-    error:function(error, context, methodName){{ returnValue = {{ {4}:'{7}', {5}:error }}; }}
+  var request = new XMLHttpRequest();
+  request.open('POST', options.serviceUrl + '/' + options.{3}, false);
+  request.setRequestHeader('Content-Type', 'application/json; charset=utf-8');
+  var failedHandler = function() {{ 
+    returnValue = {{ 
+      {4}:'{7}',
+      {5}: {{
+        readyState: request.readyState,
+        responseText: request.responseText,
+        status: request.status,
+        statusText: request.statusText
+      }}
+    }};
+  }}
+  request.onload = function() {{
+    if (request.status >= 200 && request.status <= 299)
+      returnValue = {{ {4}:'{6}', {5}: JSON.parse(request.response).d }};
+    else
+      failedHandler();
   }};
-
-  Remotion.jQuery.ajax(request);
+  request.onerror = failedHandler;
+  request.send(data);
 
   // Normalize communication between JS and C# API: always return an array if succeeded.
-  if(returnValue.state === 'success' && !Array.isArray(returnValue.data))
-    returnValue.data = [ returnValue.data ];
+  if(returnValue.{4} === '{6}' && !Array.isArray(returnValue.{5}))
```

Comment:
This feels kind of awkward. since you're touching thsi anway, should we switch to stringinterprolation instead of the placeholder?